vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/InitialHandshakeCommandCodec.java

```diff
@@ -47,6 +47,12 @@
     @Override
     void encode(DB2Encoder encoder) {
         super.encode(encoder);
+        encoder.socketConnection.closeHandler(h -> {
+          if (status == ST_CONNECTING) {
+            //Sometimes DB2 closes the connection when sending an invalid Database name.
+            cmd.fail(new Throwable("Socket closed during connection"));           
```

Comment:
should this be a `DB2Exception` instead of `Throwable`? Also, in most cases when this handler runs this `InitialHandshakeCommand` will already be completed, so calling `cmd.fail()` on a command that previously completed normally seems odd